src/components/MediaList/BaseMediaList.js

```diff
@@ -120,7 +121,15 @@ function BaseMediaList({
     const isItemLoaded = (index) => media[index] != null;
     const loadMoreItems = (start) => {
       const page = Math.floor(start / 50);
-      onRequestPage(page);
+      if (inFlightPageRequests.current[page]) return Promise.resolve(null);
+      inFlightPageRequests.current[page] = 1;
+
+      return onRequestPage(page).finally(() => {
+        // an attempt to throttle since cached requests are too fast
```

Comment:
just adding some more explanation to this 👇🏻 
```suggestion
        // Without the timeout we can still get duplicate requests.
        // That is *probably* because a rerender is triggered by some
        // redux action on request completion, just *before* the new
        // playlist items are actually stored in state.
```